agent/consul/config.go

```diff
@@ -116,6 +118,11 @@ type Config struct {
 	// configured at this point.
 	NotifyListen func()
 
+	// GetNetRPCInterceptorFunc, if not nil, sets the net/rpc rpc.ServerServiceCallInterceptor on
+	// the server side to record metrics around the RPC requests. If nil, no interceptor is added to
+	// the rpc server.
+	GetNetRPCInterceptorFunc func(recorder *middleware.RequestRecorder) rpc.ServerServiceCallInterceptor
```

Comment:
I don't think I have an opinion either way but did you consider putting this in the Deps instead of the Config?

*Most* of whats in Config are simple values and not types/funcs (one exception being the `NotifyListen` func above here).

Then again, I don't know how applicable it is with what else is in Deps. What do you think?

Reply:
That's a fair point. Chatting with Dhia more, I want to move this to `Deps` and I think it should work better ✍🏼